include/zephyr/toolchain/common.h

```diff
@@ -147,9 +157,14 @@
  * not used).
  */
 #ifndef ZTEST_UNITTEST
+#ifndef __ZPP__
 #define __syscall static inline
 #define __syscall_always_inline static inline __attribute__((always_inline))
 #else
+#define __syscall __zpp(func("syscall", __FILE__, __LINE__))
+#define __syscall_always_inline __zpp(func("syscall", __FILE__, __LINE__))
```

Comment:
question: both expand to the same?

Reply:
The `__syscall` part hasn't been refactored yet, and this is mostly to demonstrate part of the output.

But I think the current syscall parsing also behaves the same, they end up different when compiling only.